### PR TITLE
[Feature] Application placeholder pages

### LINKED ISSUE #6027: ✨Application Revamp: Placeholder pages with urls and breadcrumbs and steps
## ✨ Feature

Setup placeholder pages (with routing) for the Application steps.

## 🕵️ Details

The hero and stepper component are essentially the same on each step, with only minor changes to the breadcrumbs. This may be a good place to use the _Layout and <Outlet/> pattern enabled by React Router.

## 🎨 Design File

https://www.figma.com/file/BXYVsu1fo9LS2IySJDUINY/Applicants---Application?node-id=636-59839&t=kmjTtgdWWyX0ddZ9-0

## 📸 Screenshot

Image

## 🙋‍♀️ Proposed Implementation

Proposed urls for the steps (open to suggested changes):
- applications/:id/welcome
- applications/:id/profile
- applications/:id/resume-introduction
- applications/:id/resume (or /experiences?)
- applications/:id/resume/add (or /experiences/create?)
- applications/:id/resume/:id (or /experiences/:id?)
- applications/:id/education
- applications/:id/skills-introduction
- applications/:id/skills
- applications/:id/questions-introduction
- applications/:id/questions
- applications/:id/review
- applications/:id/success

Note that the introduction pages show the same step (in the Stepper component) as the main page, all the resume/experience pages show the same step, and the success page shows no "current" step.

## 🌎 Localization

(optional) Provide any new copy along with translations available

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the feature was properly implemented.

- [ ] All the application step pages exist under their own urls
- [ ] Pages are accessible only when the feature flag "Application Revamp" is ON 
- [ ] Stepper on each page shows the correct current page
- [ ] Each page displays the Hero, breadcrumbs, and Stepper component, with the main content left empty for future stories.

## :stop_sign: Blockers
Blocked by #5875

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #6027 

## 👋 Introduction

Adds the placeholder pages within a layout component with `Hero`, `Breadcrumbs` and `Stepper` component.

## 📋 TO DO

- [x] Add missing translations

## 🧪 Testing

> **Note**
> In order to test this properly, you will need to add `preventDisable` on the `Stepper` component to easily go through each step unimpeded

1. Add `preventDissable` prop to `Stepper` component
2. Build app `npm run build`
3. Start an application
4. Update the url to `/applications/{applicationId}/welcome
5. Step through the pages ensuring the content that is there matches [the mock ups](https://www.figma.com/file/BXYVsu1fo9LS2IySJDUINY/Applicants---Job-application?node-id=636-59839&t=iIqEBZTRvzLwb9uT-0)

## 📸 Screenshot
localhost_8000_en_applications_58bafc9c-4760-4750-a5cb-6375a86aed43_welcome

### CODE REVIEW

**Review by tristan-orourke — COMMENTED** [review]
I got a console error when clicking around between steps.
`Warning: Cannot update a component (`ApplicationApi`) while rendering a different component (`ApplicationApi`). To locate the bad setState() call inside `ApplicationApi`, follow the stack trace as described in https://reactjs.org/link/setstate-in-render`

**Comment by esizer** [comment]
> I got a console error when clicking around between steps.
> `Warning: Cannot update a component (`ApplicationApi`) while rendering a different component (`ApplicationApi`). To locate the bad setState() call inside `ApplicationApi`, follow the stack trace as described in https://reactjs.org/link/setstate-in-render`

Were you just using the stepper to get to different pages or something else? I'm having trouble reproducing 😢

**Review by tristan-orourke — CHANGES_REQUESTED** [review]
I like your solution of putting /introduction at the end of the url as another segment. Makes sense, and makes determining the current step for the stepper nice and easy.

I think you're just missing one page for editing an existing experience - and the it needs to be added to Router as well, along with the add-experience url.

**Comment by esizer** [comment]
> I like your solution of putting /introduction at the end of the url as another segment. Makes sense, and makes determining the current step for the stepper nice and easy.
> 
> I think you're just missing one page for editing an existing experience - and the it needs to be added to Router as well, along with the add-experience url.

Oops 😅, Looks like I also missed the add page. Added in [add missing resume pages](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6086/commits/b817a8d296e69cbdaf6060d11de3859a2d7cda5b)

> I was just clicking on steps in the stepper when I got the error. But going back to reproduce, I get the error just by refreshing a single page. I suspect its a race condition which has something to do with having `<ApplicationApi>` at multiple levels. ApplicationLayout is wrapped in `<ApplicationApi>`, but so is each individual subpage.
> 
> Here's the full error log: [ApplicationApi error.log](https://github.com/GCTC-NTGC/gc-digital-talent/files/11105488/ApplicationApi.error.log)

Awesome, thanks! Turns out that is exactly what it was. I didn't realize I was wrapping the layout in it as well. Changing it to use cache first worked 👍  [update to use cache-first to avoid race-condition](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6086/commits/02c56289e5d685b955a4df76ba9f5c8709484cdd)

**Review by tristan-orourke — CHANGES_REQUESTED** [review]

File: apps/web/src/pages/Applications/ApplicationResumeAddPage/ApplicationResumeAddPage.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  const path = paths.applicationResumeAdd(application.id);
+  return {
+    title: intl.formatMessage({
+      defaultMessage: "Add an experience to your résumé",
+      id: "fBabZh",
+      description: "Page title for the application résumé add experience page",
+    }),
+    icon: StarIcon,
+    omitFromStepper: true,
+    crumbs: [
```
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/pages/Applications/ApplicationResumeAddPage/ApplicationResumeAddPage.tsx:25]: I like this idea! 👍 Do you think we should do this with introduction pages as well? 🤔

[the remark this answers is not in the document]
Comment (tristan-orourke) [reply apps/web/src/pages/Applications/ApplicationResumeAddPage/ApplicationResumeAddPage.tsx:25]: I'd leave the introduction pages as is. Going from the introduction page to the main step doesn't feel like going "back up" the hierarchy in the way implied by breadcrumbs.

File: apps/web/src/hooks/useRoutes.ts [hunk 2]
```diff
@@ -162,6 +162,57 @@ const getRoutes = (lang: Locales) => {
...
+      path.join(
+        baseUrl,
+        "applications",
+        applicationId,
+        "resume",
+        "introduction",
+      ),
+    applicationResumeAdd: (applicationId: string) =>
+      path.join(baseUrl, "applications", applicationId, "resume", "add"),
+    applicationResumeEdit: (applicationId: string, experienceId: string) =>
```
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/hooks/useRoutes.ts:181]: Rwemoved in [remove unnecessary path segment](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6086/commits/047bcf8e3ac12167b4547894ad65e3c4b6c9bb01) then fixed in the router in [remove route param from router](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6086/commits/62b21444404ec13fe64b86b580065be65bded4e9)

File: apps/web/src/pages/Applications/ApplicationResumeAddPage/ApplicationResumeAddPage.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/pages/Applications/ApplicationResumeAddPage/ApplicationResumeAddPage.tsx:25]: Sounds good, updated in [update resume page breadcrumbs](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6086/commits/a201f7ee03df29fb03b9dafabd2b79c1e8086b46)

File: apps/web/src/pages/Applications/ApplicationResumeEditPage/ApplicationResumeEditPage.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  const path = paths.applicationResumeEdit(application.id, resourceId ?? "");
+  return {
+    title: intl.formatMessage({
+      defaultMessage: "Edit your experience",
+      id: "WiUlEh",
+      description: "Page title for the application résumé edit experience page",
+    }),
+    icon: StarIcon,
+    omitFromStepper: true,
+    crumbs: [
```
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/pages/Applications/ApplicationResumeEditPage/ApplicationResumeEditPage.tsx:27]: Updated in [update resume page breadcrumbs](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6086/commits/a201f7ee03df29fb03b9dafabd2b79c1e8086b46)

**Review by tristan-orourke — APPROVED** [review]
So... The error is still there, in development mode 😢 . But, since it's not present in production mode, and this is blocking other work, I say let's merge this and make another issue for the bug.

**Comment by tristan-orourke** [comment]
https://github.com/GCTC-NTGC/gc-digital-talent/issues/6146

**Pull request merged**